argus/backend/plugins/driver_matrix_tests/model.py

```diff
@@ -0,0 +1,148 @@
+
+from dataclasses import dataclass
+from typing import TypedDict
+from datetime import datetime
+from uuid import UUID
+from cassandra.cqlengine import columns
+from argus.backend.db import ScyllaCluster
+from argus.backend.plugins.core import PluginModelBase
+from argus.backend.plugins.driver_matrix_tests.udt import TestCollection, TestSuite, TestCase, EnvironmentInfo
+
+
+class RawMatrixTestCase(TypedDict):
+    name: str
+    status: str
+    time: float
+    classname: str
+    message: str
+
+
+class RawMatrixTestSuite(TypedDict):
+    name: str
+    tests: int
+    failures: int
+    disabled: int
+    skipped: int
+    passed: int
+    errors: int
+    time: float
+    cases: list[RawMatrixTestCase]
+
+
+class RawMatrixTestResult(TypedDict):
```

Comment:
can you supply a few examples out of real run,
also where is the code that gonna create this structure ? 
won't it be easier to pass the xml files back, and Argus to parse them ?

Reply:
or at least have the client code know how todo that, otherwise it would need to be implemented again and again in multiple code bases

Reply:
So initially I thought about parsing XMLs on the framework side (via a separate module that runs post driver test), maintaining it for all 3 types due to slight differences in result xmls (at least I think they're slightly different). I think yes, if the XMLs are not in fact different at all the parsing code should be in argus instead and all driver tests would need is to launch client and send the xml file to the argus side. The structures right now are built client side and `RawMatrix...` classes are used to hint request json once it arrives.

Reply:
the XML are quite similar (even that there no exact specification for xunit they adhere to)
they are all align in a sense they were create to be visible in jenkins, so the jenkins xunit plugin read them all correctly

Reply:
Done, code is now located inside driver matrix client.